src/js/game/LevelMVC/LevelView.js

```diff
@@ -1462,6 +1497,69 @@ module.exports = class LevelView {
+
+    frameList = this.generateFramesWithEndDelay("Player_", 263, 262, "Player_262", 3, 5);
+    frameList.push("Player_263");
+    this.agent.sprite.animations.add('lookAtCam_down', frameList, idleFrameRate, false).onComplete.add(() => {
+      this.playScaledSpeed(this.agent.sprite.animations, "idlePause_down");
+    });
+
+    frameList = this.generateFramesWithEndDelay("Player_", 270, 269, "Player_269", 3, 5);
+    frameList.push("Player_270");
+    this.agent.sprite.animations.add('lookAtCam_right', frameList, idleFrameRate, false).onComplete.add(() => {
+      this.playScaledSpeed(this.agent.sprite.animations, "idlePause_right");
+    });
+
+    frameList = this.generateFramesWithEndDelay("Player_", 277, 276, "Player_276", 3, 5);
+    frameList.push("Player_277");
+    this.agent.sprite.animations.add('lookAtCam_up', frameList, idleFrameRate, false).onComplete.add(() => {
+      this.playScaledSpeed(this.agent.sprite.animations, "idlePause_up");
+    });
+
+    frameList = this.generateFramesWithEndDelay("Player_", 284, 283, "Player_283", 3, 5);
+    frameList.push("Player_284");
+    this.agent.sprite.animations.add('lookAtCam_left', frameList, idleFrameRate, false).onComplete.add(() => {
+      this.playScaledSpeed(this.agent.sprite.animations, "idlePause_left");
+    });
+
+    this.agent.sprite.animations.add('mine_down', Phaser.Animation.generateFrameNames("Player_", 241, 244, "", 3), frameRate, true);
+    this.agent.sprite.animations.add('mine_right', Phaser.Animation.generateFrameNames("Player_", 245, 248, "", 3), frameRate, true);
+    this.agent.sprite.animations.add('mine_up', Phaser.Animation.generateFrameNames("Player_", 249, 252, "", 3), frameRate, true);
+    this.agent.sprite.animations.add('mine_left', Phaser.Animation.generateFrameNames("Player_", 253, 256, "", 3), frameRate, true);
+
+    this.agent.sprite.animations.add('mineCart_down', Phaser.Animation.generateFrameNames("Minecart_", 5, 5, "", 2), frameRate, false);
+    this.agent.sprite.animations.add('mineCart_turnleft_down', Phaser.Animation.generateFrameNames("Minecart_", 6, 6, "", 2), frameRate, false);
+    this.agent.sprite.animations.add('mineCart_turnright_down', Phaser.Animation.generateFrameNames("Minecart_", 12, 12, "", 2), frameRate, false);
+
+    this.agent.sprite.animations.add('mineCart_right', Phaser.Animation.generateFrameNames("Minecart_", 7, 7, "", 2), frameRate, false);
+    this.agent.sprite.animations.add('mineCart_left', Phaser.Animation.generateFrameNames("Minecart_", 11, 11, "", 2), frameRate, false);
+
+    this.agent.sprite.animations.add('mineCart_up', Phaser.Animation.generateFrameNames("Minecart_", 9, 9, "", 2), frameRate, false);
+    this.agent.sprite.animations.add('mineCart_turnleft_up', Phaser.Animation.generateFrameNames("Minecart_", 10, 10, "", 2), frameRate, false);
+    this.agent.sprite.animations.add('mineCart_turnright_up', Phaser.Animation.generateFrameNames("Minecart_", 8, 8, "", 2), frameRate, false);
+  }
+
```

Comment:
Why is this duplicating code from `preparePlayerSprite`?  https://github.com/code-dot-org/craft/blob/8fa65f72d375f4d33aa376a99bcff3a84fc9a920/src/js/game/LevelMVC/LevelView.js#L1434-L1498

Reply:
Setting up the the animations for this.agent?

this.agent.sprite = this.actionPlane.create(0, 0, `player${playerName}`, 'Player_121');
^With the string playerAgent it should be referencing the agent JSON ~> Agent png and then adding all the sprites (which have the same names as the Alex/Steve sprites I believe) to the agent's sprite.animations.

That was my objective anyway.

Reply:
Also @joshlory aren't you on vacation? Why are you responding to work stuff? 😂

Reply:
Dammit Josh, go relax!

Reply:
:smile: :palm_tree:

Reply:
Right now preparePlayerSprie handles both cases (Agent, Steve, Alex, SteveEvents, AlexEvents). We shouldn't need to duplicate the code in order to keep the Agent working.

Reply:
Yeah, the logic in preparePlayerSprite logic is 100% covered, but it directly references this.player (which is just levelModel.player) when assigning sprites. If we're going to have a whole second entity we'll need to be able to assign sprites to that entity as well. I'm making the function generalized this morning to take an optional entity argument (default to this.player) that should eliminate the need to make a duplicate function 👍